improver/precipitation_type/snow_splitter.py

```diff
@@ -0,0 +1,159 @@
+        )
+
+        return (rain_cube, snow_cube, precip_cube)
+
+    def process(self, cubes: CubeList,) -> Cube:
+        """
+        Splits the precipitation cube data into a snow or rain contribution.
+
+        Whether the output is a rate or accumulation will depend on the precipitation_cube.
+        output_is_rain will determine whether the outputted cube is a cube of snow or rain.
+
+        The probability of rain and snow at the surfaces should only contain 1's where the
+        phase is present at the surface and 0's where the phase is not present
+        at the surface. These cubes need to be consistent with each other such that both
+        rain and snow can't be present at the surface (e.g. at no grid square can both
+        diagnostics have a probability of 1).
+
+        A grid of coefficients is calculated by an arbitrary function that maps
+        (0,0) -> 0.5, (1,0) -> 1, (0,1) -> 0 where the first coordinate is the probability
+        for the variable that will be outputted. This grid of coefficients is then multiplied
+        by the precipitation rate/accumulation to split out the contribution of the desired
+        variable.
+
+        Args:
+            cubes:
+                containing:
+                    rain_cube:
+                        Cube of the probability of rain at the surface.
+                    snow_cube:
+                        Cube of the probability of snow at the surface.
+                    precip_cube:
+                        Cube of either precipitation rate or precipitation accumulation.
+
+        Returns:
+            Cube of rain/snow (depending on self.output_is_rain) rate/accumulation (depending on
+            precipitation cube)
+
+        Raises:
+            ValueError: If, at some grid square, both snow_cube and rain_cube have a probability of
+            0
+        """
+
+        rain_cube, snow_cube, precip_cube = self.separate_input_cubes(cubes)
+
+        assert_spatial_coords_match([rain_cube, snow_cube, precip_cube])
+        if np.any((rain_cube.data + snow_cube.data) == 2):
+            raise ValueError(
+                """There is at least 1 grid square where the probability of snow
+                             at the surface and the probability of rain at the surface are both
+                             1."""
+            )
+
+        if self.output_is_rain:
+            required_cube = rain_cube
+            other_cube = snow_cube
+            name = "rain"
+        else:
+            required_cube = snow_cube
+            other_cube = rain_cube
+            name = "snow"
+
+        # arbitrary function that maps combinations of rain and snow probabilities
+        # to an appropriate coefficient
+        coefficient_cube = (required_cube - other_cube + 1) / 2
+        coefficient_cube.data = coefficient_cube.data.astype(np.float32)
+
+        new_name = precip_cube.name().replace("precipitation", name)
```

Comment:
This needs to be a bit smarter. The CF names are:
- "lwe_precipitation_rate"
- "lwe_snow_rate"
- "rain_rate"
Therefore we need to remove the "lwe_" when outputting rain.

Reply:
I've updated the code to remove the lwe part and updated the unit tests to give the precip rate cube the "lwe_" cube name so this is tested. I've recreated the acceptance tests kgo as well